lib/diagnostic.js

```diff
@@ -50,8 +53,8 @@ const last = function (array, callback) {
 // for one and only one element
 // you may use any feature of JavaScript
 // to solve this problem
-const hasOne = function (array, callback) {
-
+const hasOne = arr.filter((array, callback)) {
+  return true
```

Comment:
You want to make sure that the `filter`ed array is the same size as the original here